### PR TITLE
Updated ItemRegistryImpl to make use of {@inheritDoc} notation

### PR DESCRIPTION
**Pull request by AlexanderKostadinov28**
Removed `{@inheritDoc}` and `(non-Javadoc) @see` notations throughout ESH completely (code style consistency).

Signed-off-by: Alexander Kostadinov <alexander.g.kostadinov@gmail.com>

### CODE REVIEW

**Comment by maggu2810** [comment]
Reading the JavaDoc documentation (http://docs.oracle.com/javase/8/docs/technotes/tools/windows/javadoc.html) states that inheritDoc is already done implicit for overwritten methods.

So I would prefer not to add implicit stuff as it does not improve readability of the code at all.

**Comment by maggu2810** [comment]
So, I agree with removing the JavaDoc for overwritten methods if it is equal to the one of the parent class but remove the use of inheritDoc.

**Comment by AlexanderKostadinov28** [comment]
I agree then, but in such case we should remove the JavaDoc for all inherited methods from `ItemRegistry` (there are some left `non-JavaDoc @see` notations).

**Comment by kaikreuzer** [comment]
> remove the use of inheritDoc

I agree and that is also the style we tried to apply throughout ESH so far.

**Comment by AlexanderKostadinov28** [comment]
In such case, I can remove the {@inheritDoc} notation, as well as get rid of unnecessary javadoc notations for overriden methods throughout the whole ESH.

**Comment by AlexanderKostadinov28** [comment]
Finally, I removed {@inheritDoc} and (non-Javadoc) @see notations throughout ESH completely for code style consistency. Significant number of the (non-Javadoc) @see notations were obsolete and mentioned wrong location.

File: bundles/model/org.eclipse.smarthome.model.item/src/org/eclipse/smarthome/model/item/internal/GenericItemProvider.java [hunk 1]
```diff
@@ -340,8 +337,6 @@ private void internalDispatchBindings(BindingConfigReader reader, String modelNa
     }
 
     /**
-     * {@inheritDoc}
-     * <p>
```
Comment (maggu2810) [inline bundles/model/org.eclipse.smarthome.model.item/src/org/eclipse/smarthome/model/item/internal/GenericItemProvider.java]: I am not sure about this one.
Wasn't the previous approach to use the parent JavaDoc and additionally an sentence?
Isn't now the parent JavaDoc overwritten by the sentence?

I can't check it ATM but it feels like "parent + custom" before and "custom" now. WDYT?

File: extensions/transform/org.eclipse.smarthome.transform.scale/src/main/java/org/eclipse/smarthome/transform/scale/internal/ScaleTransformationService.java [hunk 2]
```diff
@@ -47,9 +47,6 @@
      *            the list of properties defining all the available ranges
      * @param source
      *            the input to transform
-     *
```
Comment (maggu2810) [inline extensions/transform/org.eclipse.smarthome.transform.scale/src/main/java/org/eclipse/smarthome/transform/scale/internal/ScaleTransformationService.java]: This JavaDoc seems to be wrong formatted for me at all.

So, not really part of your "remove inheritDoc" changes, but I don't think a correct JavaDoc starts with a `<p>` at all.
Shouldn't it start with a short sentence and then perhaps a longer description.

Another todo that we should remember is the removal of all `</p>`. AFAIK there is only `<p>` defined for "correct" JavaDoc.

File: extensions/transform/org.eclipse.smarthome.transform.xslt/src/main/java/org/eclipse/smarthome/transform/xslt/internal/XsltTransformationService.java [hunk 3]
```diff
@@ -46,9 +46,6 @@
      *            as well
      * @param source
      *            the input to transform
-     *
```
Comment (maggu2810) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/src/main/java/org/eclipse/smarthome/transform/xslt/internal/XsltTransformationService.java]: See above (don't start with `<p>`, remove `</p>`)

File: bundles/model/org.eclipse.smarthome.model.item/src/org/eclipse/smarthome/model/item/internal/GenericItemProvider.java (hunk 1 above)
Comment (AlexanderKostadinov28) [inline bundles/model/org.eclipse.smarthome.model.item/src/org/eclipse/smarthome/model/item/internal/GenericItemProvider.java]: Actually, you are right - there is no JavaDoc in the parent but only in the inherited class.  So, I will update the PR and remove the JavaDoc from here and move it to ModelRepositoryChangeListener.

File: extensions/transform/org.eclipse.smarthome.transform.scale/src/main/java/org/eclipse/smarthome/transform/scale/internal/ScaleTransformationService.java (hunk 2 above)
Comment (AlexanderKostadinov28) [inline extensions/transform/org.eclipse.smarthome.transform.scale/src/main/java/org/eclipse/smarthome/transform/scale/internal/ScaleTransformationService.java]: Here I will remove the <p> in the beginning, as well as edit the JavaDoc according to your comment. In general, in the upcoming days, I will create another PR for fixing all of the unclosed <p> tags in JavaDocs throughout ESH - this has to be done. :)

File: extensions/transform/org.eclipse.smarthome.transform.xslt/src/main/java/org/eclipse/smarthome/transform/xslt/internal/XsltTransformationService.java (hunk 3 above)
Comment (AlexanderKostadinov28) [inline extensions/transform/org.eclipse.smarthome.transform.xslt/src/main/java/org/eclipse/smarthome/transform/xslt/internal/XsltTransformationService.java]: I will do the same as above.

**Comment by AlexanderKostadinov28** [comment]
@maggu2810 All of your latest comments are resolved now. As said, I will consider creating another PR for fixing the JavaDoc </p> usage.

**Pull request merged**